drivers/interrupt_controller/intc_gic.c

```diff
@@ -55,23 +55,23 @@ bool arm_gic_irq_is_enabled(unsigned int irq)
 void arm_gic_irq_set_priority(
 	unsigned int irq, unsigned int prio, u32_t flags)
 {
-	int int_grp, int_off;
-	u8_t val;
+	unsigned int int_grp, int_off;
```

Comment:
This change from `int` to `unsigned int` is not necessary for fixing the issue and should not be part of this patch.

In this particular instance, it does not really matter if we use `signed int` or `unsigned int` for `int_grp` and `int_off`; though, it would semantically make more sense to use an unsigned type (i.e. `u32_t` for Zephyr) as you suggest.

If you would like to introduce this change, please do so in a separate commit, for all functions in this driver that have `int_grp` and `int_off`-type variables.